Back out "Revert D49107540: [pytorch][PR] split by tag"

strisunshinewentingwang opened the pull request:
Summary:
Original commit changeset: 6391a068640b

Original Phabricator Diff: D49107540

Test Plan: same as D49107540

Differential Revision: D49297522

clee2000 commented on test/fx/test_fx_split.py:
```diff
@@ -30,3 +34,115 @@ def forward(self, x, y):
                     self.assertEqual(n.meta["name"], n.name)
+
+
+class TestSplitByTags(unittest.TestCase):
```
This should be using the TestCase imported from torch.testing._internal.common_utils, not unittest

Commit:
Back out "Revert D49107540: [pytorch][PR] split by tag" (#109332)

Summary:

Original commit changeset: 6391a068640b

Original Phabricator Diff: D49107540

bypass-github-pytorch-ci-checks


FB: https://fb.workplace.com/groups/1405155842844877/permalink/7432758670084534/

Test Plan: same as D49107540

Reviewed By: uciyc123, 842974287

Differential Revision: D49297522
